server/player/player.go

```diff
@@ -1835,22 +1835,42 @@ func (p *Player) BreakBlock(pos cube.Pos) {
 }
 
 // drops returns the drops that the player can get from the block passed using the item held.
-func (p *Player) drops(held item.Stack, b world.Block) []item.Stack {
+func (p *Player) drops(held item.Stack, b world.Block, pos cube.Pos) []item.Stack {
 	t, ok := held.Item().(item.Tool)
 	if !ok {
 		t = item.ToolNone{}
 	}
 	var drops []item.Stack
 	if container, ok := b.(block.Container); ok {
-		// If the block is a container, it should drop its inventory contents regardless whether the
-		// player is in creative mode or not.
-		drops = container.Inventory().Items()
+		if c, ok := b.(block.Chest); ok && c.Paired() {
+			pairPos := c.PairPos(pos)
+			left, right := pos, pairPos
+			if pos.Side(c.Facing.RotateRight().Face()) == pairPos {
+				left, right = right, left
+			}
+
+			for slot, i := range c.Inventory().Slots() {
+				if i.Empty() {
+					continue
+				}
+
+				if slot < 27 && pos == left {
+					drops = append(drops, i)
+				} else if slot > 26 && pos == right {
+					drops = append(drops, i)
+				}
+			}
+		} else {
+			// If the block is a container, it should drop its inventory contents regardless whether the
+			// player is in creative mode or not.
+			drops = container.Inventory().Items()
+			container.Inventory().Clear()
+		}
```

Comment:
Does this code for the drops need to be here? Can we not put this in the BreakInfo of Chest?

Reply:
it does not i just had it here because this is where the other container drops were